### ISSUE TITLE
1.1.17: Segfault after rejecting listener

### ISSUE THREAD

**Issue by mandarjog** [istio/istio#18078]
```
2019-10-19T07:48:22.663786Z	info	Version root@a3952397-ef79-11e9-842f-2a0912242356-docker.io/istio-1.1.17-149c2fc74f0af60e8dedb6d69d58f1bbab3e0ec4-Clean
2019-10-19T07:48:22.663919Z	info	Obtained private IP [10.36.7.54]
2019-10-19T07:48:22.663969Z	info	Proxy role: &model.Proxy{ClusterID:"", Type:"router", IPAddresses:[]string{"10.36.7.54", "10.36.7.54"}, ID:"istio-ingressgateway-56f94b9b4b-kfxtl.istio-system", Locality:(*core.Locality)(nil), DNSDomain:"istio-system.svc.cluster.local", ConfigNamespace:"", TrustDomain:"cluster.local", Metadata:map[string]string{}, SidecarScope:(*model.SidecarScope)(nil), ServiceInstances:[]*model.ServiceInstance(nil), WorkloadLabels:model.LabelsCollection(nil)}
2019-10-19T07:48:22.663979Z	info	PilotSAN []string(nil)
2019-10-19T07:48:22.664873Z	info	Effective config: binaryPath: /usr/local/bin/envoy
configPath: /etc/istio/proxy

2019-10-19T07:50:04.985595Z     info    Envoy proxy is ready
[2019-10-19 07:50:31.292][19][warning][config] [bazel-out/k8-opt/bin/external/envoy/source/common/config/_virtual_includes/grpc_mux_subscription_lib/common/config/grpc_mux_subscription_impl.h:77] gRPC config for type.googleapis.com/envoy.api.v2.Listener rejected: Error adding/updating listener(s) 0.0.0.0_15443: cannot bind '0.0.0.0:15443': Address already in use, 0.0.0.0_80: cannot bind '0.0.0.0:80': Address already in use
[2019-10-19 07:51:04.637][19][warning][main] [external/envoy/source/server/server.cc:425] caught SIGTERM
```

```c++
[2019-10-19 07:51:04.637][19][warning][main] [external/envoy/source/server/server.cc:425] caught SIGTERM
[2019-10-19 07:51:04.670][19][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:81] Caught Segmentation fault, suspect faulting address 0x3d57700
[2019-10-19 07:51:04.670][19][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:69] Backtrace (use tools/stack_decode.py to get line numbers):
[2019-10-19 07:51:04.670][19][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #0: __restore_rt [0x7fc91453e390]
[2019-10-19 07:51:04.675][19][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #1: Envoy::Config::GrpcMuxImpl::GrpcMuxWatchImpl::~GrpcMuxWatchImpl() [0xd54701]
[2019-10-19 07:51:04.680][19][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #2: Envoy::Config::GrpcMuxImpl::GrpcMuxWatchImpl::~GrpcMuxWatchImpl() [0xd5479e]
[2019-10-19 07:51:04.684][19][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #3: Envoy::Config::GrpcSubscriptionImpl::~GrpcSubscriptionImpl() [0xbf514a]
[2019-10-19 07:51:04.689][19][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #4: Envoy::Secret::SdsApi::~SdsApi() [0xc2d66f]
[2019-10-19 07:51:04.693][19][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #5: Envoy::Extensions::TransportSockets::Tls::ContextConfigImpl::~ContextConfigImpl() [0xbff7ad]
[2019-10-19 07:51:04.697][19][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #6: Envoy::Extensions::TransportSockets::Tls::ClientContextConfigImpl::~ClientContextConfigImpl() [0xc007f3]
[2019-10-19 07:51:04.701][19][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #7: Envoy::Extensions::TransportSockets::Tls::ClientSslSocketFactory::~ClientSslSocketFactory() [0x8aa282]
[2019-10-19 07:51:04.705][19][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #8: Envoy::Upstream::ClusterInfoImpl::~ClusterInfoImpl() [0xd45792]
[2019-10-19 07:51:04.709][19][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #9: Envoy::Upstream::ClusterInfoImpl::~ClusterInfoImpl() [0xd457ee]
[2019-10-19 07:51:04.714][19][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #10: Envoy::Upstream::HostDescriptionImpl::~HostDescriptionImpl() [0xa6277b]
[2019-10-19 07:51:04.718][19][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #11: Envoy::Upstream::HostImpl::~HostImpl() [0xd44e87]
[2019-10-19 07:51:04.722][19][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #12: std::__1::__shared_ptr_pointer<>::__on_zero_shared() [0xc4d849]
[2019-10-19 07:51:04.726][19][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #13: Envoy::Upstream::HostSetImpl::~HostSetImpl() [0xc6df7b]
[2019-10-19 07:51:04.730][19][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #14: Envoy::Upstream::HostSetImpl::~HostSetImpl() [0xd4546e]
[2019-10-19 07:51:04.734][19][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #15: Envoy::Upstream::PrioritySetImpl::~PrioritySetImpl() [0xc4c72b]
[2019-10-19 07:51:04.737][19][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #16: Envoy::Upstream::ClusterManagerImpl::ThreadLocalClusterManagerImpl::ClusterEntry::~ClusterEntry() [0xc4c89e]
[2019-10-19 07:51:04.741][19][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #17: Envoy::Upstream::ClusterManagerImpl::ThreadLocalClusterManagerImpl::~ThreadLocalClusterManagerImpl() [0xc49fbc]
[2019-10-19 07:51:04.746][19][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #18: Envoy::ThreadLocal::InstanceImpl::shutdownThread() [0xbc4779]
[2019-10-19 07:51:04.750][19][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #19: Envoy::Server::InstanceImpl::terminate() [0xbca921]
[2019-10-19 07:51:04.754][19][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #20: Envoy::Server::InstanceImpl::run() [0xbce2f6]
[2019-10-19 07:51:04.758][19][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #21: Envoy::MainCommonBase::run() [0x88fc19]
[2019-10-19 07:51:04.762][19][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #22: main [0x88edc2]
[2019-10-19 07:51:04.762][19][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:75] #23: [0x7fc91377f830]
2019-10-19T07:51:05.099554Z	warn	Epoch 0 terminated with an error: signal: segmentation fault (core dumped)
```

**Comment by lambdai** [comment]
known issue across all istio versions

Definitely be fixed in 1.5:
Envoy crashes at terminate(). the latest update is https://github.com/envoyproxy/envoy/pull/8553
To avoid envoy restart spaghetti sequences: use SDS in gateway. I believe @costinm is working on it

**Comment by lambdai** [comment]
Update:
sds ingress is enabled since 1.5
Wont work on envoy side
